pkg/providers/launchtemplate/launchtemplate.go

```diff
@@ -156,24 +164,37 @@ func (p *Provider) getStaticParameters(ctx context.Context, instanceType *cloudp
 		NetworkPlugin:                  options.FromContext(ctx).NetworkPlugin,
 		NetworkPolicy:                  options.FromContext(ctx).NetworkPolicy,
 		SubnetID:                       subnetID,
+		ClusterResourceGroup:           p.clusterResourceGroup,
 	}, nil
 }
 
-func (p *Provider) createLaunchTemplate(options *parameters.Parameters) (*Template, error) {
-	// render user data
-	userData, err := options.UserData.Script()
-	if err != nil {
-		return nil, err
-	}
-
+func (p *Provider) createLaunchTemplate(ctx context.Context, params *parameters.Parameters) (*Template, error) {
 	// merge and convert to ARM tags
-	azureTags := mergeTags(options.Tags, map[string]string{karpenterManagedTagKey: options.ClusterName})
+	azureTags := mergeTags(params.Tags, map[string]string{karpenterManagedTagKey: params.ClusterName})
 	template := &Template{
-		UserData: userData,
-		ImageID:  options.ImageID,
-		Tags:     azureTags,
-		SubnetID: options.SubnetID,
+		ImageID:        params.ImageID,
+		Tags:           azureTags,
+		SubnetID:       params.SubnetID,
+		IsWindows:      params.IsWindows,
+		StorageProfile: params.StorageProfile,
 	}
+
+	if p.provisionMode == options.ProvisionModeBootstrappingClient {
+		customData, cse, err := params.AgentBakerNodeBootstrapping.GetCustomDataAndCSE(ctx)
+		if err != nil {
+			return nil, err
+		}
+		template.AgentBakerCustomData = customData
+		template.AgentBakerCSE = cse
+	} else {
+		// render user data
+		userData, err := params.SelfContainedUserData.Script()
```

Comment:
is self contained ever using userdata? why is it called this? Maybe im misunderstanding something?